doc/source/api.rst

```diff
@@ -1834,6 +1866,16 @@ Conversion
    DatetimeIndex.to_pydatetime
    DatetimeIndex.to_series
    DatetimeIndex.to_frame
+   DatetimeIndex.shift
+
+Sorting
+~~~~~~~
+.. autosummary::
+   :toctree: generated/
+
+   DatetimeIndex.argsort
+   DatetimeIndex.searchsorted
+   DatetimeIndex.sort_values
```

Comment:
I don't think they should be showing up in the DatetimeIndex class summary, since they aren't explicitly listed there under `Methods`.

Reply:
we have links in the whatsnew that need references, otherwise they don't go anywhere, e.g. ``:func:`DatetimeIndex.searchsorted` ``

Reply:
Does adding them to the api.rst fix the references? Or do you need to also add them to the methods section in the DatetimeIndex docstring?

An alternative is something like "Fixed bug in :meth:\`Index.searchsorted\` for indexes with datetime values." would work.

Reply:
adding to api.rst fixes this.

> Does adding them to the api.rst fix the references? Or do you need to also add them to the methods section in the DatetimeIndex docstring?

What exactly do we guarantee with this? (answer is I am not sure)

> "Fixed bug in :meth:`Index.searchsorted` 

this would work as well. But, we currently don't have a method to check this (except visually).

Reply:
After the release I'll dust off https://github.com/pandas-dev/pandas/pull/22743, which will fail the doc build if there are any warnings. So these kind of linking to a method that isn't in our API docs would have been caught when the PR was made.

Reply:
yep

Reply:
ok I reverted most of the additions and changed the refs. To be clear, I think this is actually a bug in that an invalid ``:meth:`` or ``:func:`` is not a warning if its misformatted, but I also think it is not warning when we just a sub-class of Index, but it doesn't render (the link). @datapythonista